server/models/user.model.js

```diff
@@ -1,29 +1,33 @@
-/**
- * User Schema
- */
-module.exports = (sequelize, DataTypes) => {
-    const User = sequelize.define('User', {
-        id: {
-            type: DataTypes.INTEGER,
-            autoIncrement: true,
-            primaryKey: true,
+import mongoose from 'mongoose';
+
+const UserSchema = new mongoose.Schema({
+    uuid: {
+        type: String,
+        required: true,
+        unique: true,
+        index: true,
+    },
+    username: {
+        type: String,
+        required: true,
+    },
+    devices: [new mongoose.Schema({
+        type: {
+            type: String,
+            enum: ['iOS', 'android'],
+            required: true,
         },
-        uuid: {
-            type: DataTypes.UUID,
-            allowNull: false,
-            unique: true,
+        token: {
+            type: String,
+            required: true,
         },
-        username: {
-            type: DataTypes.STRING,
-            allowNull: false,
+        status: {
+            type: String,
+            enum: ['enabled', 'disabled'],
```

Comment:
how do you feel about this being an enum vs this being a bool?
For context, this exists to enable a "soft delete".
We could also just remove "soft delete" functionality and actually delete device tokens.

Reply:
Strong opinion: I'd keep soft delete ==> They don't actually want the notifications deleted from the DB for audit.
Weak opinion: I'd keep enum. More flexibility going forwards.